app/javascript/packs/runtimeBanner.jsx

```diff
@@ -8,25 +9,15 @@ function loadElement() {
   }
 }
 
-function initializeBannerWhenPageIsReady() {
-  setTimeout(() => {
-    if (document.body.getAttribute('data-loaded') === 'true') {
-      // We're ready to initialize
-      window.InstantClick.on('change', () => {
-        loadElement();
-      });
-
-      loadElement();
-    } else {
-      // Page hasn't initialized yet. We need to wait until the page is ready
-      initializeBannerWhenPageIsReady();
-    }
-  }, 100);
-}
-
 // This pack relies on the same logic as `packs/listings` & `packs/Chat`. The
 // banner lives in every page (including the main feed) and a race condition
 // occurs when the page initializes for the first time or when signing out (no
 // cache request). In order to avoid this we defer the initialization until the
 // page is actually ready.
-initializeBannerWhenPageIsReady();
+waitForDataLoaded().then(() => {
```

Comment:
I added the `.catch` clause on both places we use this 👍 

> Feedback to the user would be helpful as well.

Generally I'd agree, but these two situations maybe aren't critical in the sense they require the user to be notified something failed.

If we fail here it means the base data never loaded and other more critical areas of the app wouldn't work either (unable to post comments, reactions, etc) so there's likely other feedback the user would see to know something's wrong. These failures would cause the runtime banner not to load and some features wouldn't work as expected within the mobile apps (because of the context specific functions not being available).

If this is still something we want (to provide feedback in this `catch` clause) let me know and I could maybe add a SnackBar message saying `Something went wrong: Try reloading the page` or something along those lines. Also open to any other specific suggestion.

> Nitpick (non-blocking): Prefer async/await

I wrangled with async/await for the use of `request` here (and a few other situations before) and lost every time 😆 I'm totally down for pairing to refactor this to async/await to learn more about it afterwards